### PR TITLE
teyjus: unstable-2019-07-26 -> 2.1.1

### PR DESCRIPTION
**Pull request by bcdarwin**
###### Description of changes

Routine update; cleanup; convert from unstable to stable version; omake -> dune conversion; added changelog.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/compilers/teyjus/default.nix [hunk 1]
```diff
@@ -1,42 +1,31 @@
-{ lib, stdenv, fetchFromGitHub, omake, ocaml, flex, bison }:
+{ lib, fetchFromGitHub, buildDunePackage, flex, bison }:
 
-stdenv.mkDerivation rec {
+buildDunePackage rec {
   pname = "teyjus";
-  version = "unstable-2019-07-26";
+  version = "2.1.1";
+
+  minimalOCamlVersion = "4.05";
```
Comment (vbgl) [inline pkgs/development/compilers/teyjus/default.nix:7]: What’s the purpose of this line?

File: pkgs/development/compilers/teyjus/default.nix [hunk 2]
```diff
@@ -1,42 +1,31 @@
...
+  duneVersion = "3";
```
Comment (vbgl) [inline pkgs/development/compilers/teyjus/default.nix:8]: This line is redundant.

Comment (bcdarwin) [reply pkgs/development/compilers/teyjus/default.nix]: Removed both of these lines.

File: pkgs/development/compilers/teyjus/default.nix (hunk 1 above)
Comment (bcdarwin) [reply pkgs/development/compilers/teyjus/default.nix]: Good point, removed.

**Pull request merged**